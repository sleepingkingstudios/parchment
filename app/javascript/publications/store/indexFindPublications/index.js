import FindManyEndpoint from '../../../api/findMany';

const REQUEST_URL = '/api/publications';
const endpoint = new FindManyEndpoint({
  data: { publications: [] },
  middleware: [],
  namespace: 'publications/indexFindPublications',
  url: REQUEST_URL,
});

export default endpoint;

export const {
  actions,
  hooks,
  namespace,
  reducer,
  request,
} = endpoint;
