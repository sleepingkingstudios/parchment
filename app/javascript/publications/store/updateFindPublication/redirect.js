import { push } from 'connected-react-router';

export default {
  handleFailure: next => ({ dispatch, getState, response }) => {
    next({ dispatch, getState, response });

    dispatch(push('/publications'));
  },
};
