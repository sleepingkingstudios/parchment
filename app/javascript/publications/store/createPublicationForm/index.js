import FormEndpoint from '../../../api/form';
import { buildPublication } from '../../entities';
import alerts from './alerts';
import redirect from './redirect';

const REQUEST_URL = '/api/publications';
const endpoint = new FormEndpoint({
  data: { publication: buildPublication() },
  middleware: [
    redirect,
    alerts,
  ],
  namespace: 'publications/createPublicationForm',
  url: REQUEST_URL,
});

export default endpoint;

export const {
  actions,
  hooks,
  namespace,
  reducer,
  request,
} = endpoint;
