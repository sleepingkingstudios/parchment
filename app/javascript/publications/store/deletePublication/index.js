import DeleteOneEndpoint from '../../../api/deleteOne';
import alerts from './alerts';
import redirect from './redirect';

const REQUEST_URL = '/api/publications/:id';
const endpoint = new DeleteOneEndpoint({
  data: {},
  middleware: [
    alerts,
    redirect,
  ],
  namespace: 'publications/deletePublication',
  url: REQUEST_URL,
});

export default endpoint;

export const {
  actions,
  hooks,
  namespace,
  reducer,
  request,
} = endpoint;
