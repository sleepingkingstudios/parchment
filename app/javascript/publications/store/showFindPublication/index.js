import FindOneEndpoint from '../../../api/findOne';
import alerts from './alerts';
import { buildPublication } from '../../entities';
import redirect from './redirect';

const REQUEST_URL = '/api/publications/:id';
const endpoint = new FindOneEndpoint({
  data: { publication: buildPublication() },
  middleware: [
    redirect,
    alerts,
  ],
  namespace: 'publications/showFindPublication',
  url: REQUEST_URL,
});

export default endpoint;

export const {
  actions,
  hooks,
  namespace,
  reducer,
  request,
} = endpoint;
