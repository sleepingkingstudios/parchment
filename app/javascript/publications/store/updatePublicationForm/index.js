import FormEndpoint from '../../../api/form';
import { buildPublication } from '../../entities';
import alerts from './alerts';
import redirect from './redirect';

const namespace = 'publications/updatePublicationForm';
const REQUEST_URL = '/api/publications/:id';

export const {
  actions,
  hooks,
  reducer,
  request,
} = new FormEndpoint({
  data: { publication: buildPublication() },
  method: 'PATCH',
  middleware: [
    redirect,
    alerts,
  ],
  namespace,
  url: REQUEST_URL,
});
