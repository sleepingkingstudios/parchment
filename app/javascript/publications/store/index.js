import { combineReducers } from 'redux';

import { reducer as createPublicationForm } from './createPublicationForm';
import { reducer as indexFindPublications } from './indexFindPublications';
import { reducer as showFindPublication } from './showFindPublication';

export default combineReducers({
  createPublicationForm,
  indexFindPublications,
  showFindPublication,
});
