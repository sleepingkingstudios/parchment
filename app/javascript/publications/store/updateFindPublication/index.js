import FindOneEndpoint from '../../../api/findOne';
import alerts from './alerts';
import redirect from './redirect';
import { buildPublication } from '../../entities';
import { actions as formActions } from '../updatePublicationForm';

const REQUEST_URL = '/api/publications/:id';
const endpoint = new FindOneEndpoint({
  data: { publication: buildPublication() },
  middleware: [
    {
      handleSuccess: next => ({ dispatch, getState, response }) => {
        next({ dispatch, getState, response });

        const { publications } = getState();
        const { updateFindPublication } = publications;
        const { data } = updateFindPublication;
        const { setFormData } = formActions;

        dispatch(setFormData(data));
      },
    },
    redirect,
    alerts,
  ],
  namespace: 'publications/showFindPublication',
  url: REQUEST_URL,
});

export default endpoint;

export const {
  actions,
  hooks,
  namespace,
  reducer,
  request,
} = endpoint;
