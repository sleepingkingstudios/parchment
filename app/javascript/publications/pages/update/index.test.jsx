import React from 'react';
import { shallow } from 'enzyme';

import UpdatePublicationBreadcrumbs from './breadcrumbs';
import UpdatePublicationPage from './index';
import { hooks } from '../../store/updateFindPublication';

jest.mock('../../store/updateFindPublication');

hooks.useRequestData.mockImplementation(() => () => {});

describe('UpdatePublicationPage', () => {
  const id = '00000000-0000-0000-0000-000000000000';
  const match = { params: { id } };
  const defaultProps = { match };

  it('should render the Page', () => {
    const rendered = shallow(<UpdatePublicationPage {...defaultProps} />);

    expect(rendered).toHaveDisplayName('Page');
    expect(rendered).toHaveClassName('page-update-publication');
    expect(rendered).toHaveProp('breadcrumbs', (<UpdatePublicationBreadcrumbs />));
  });

  it('should render the heading', () => {
    const rendered = shallow(<UpdatePublicationPage {...defaultProps} />);
    const heading = rendered.find('h1');

    expect(heading).toExist();
    expect(heading).toIncludeText('Update Publication');
  });

  it('should render the publication form', () => {
    const rendered = shallow(<UpdatePublicationPage {...defaultProps} />);

    expect(rendered).toContainMatchingElement('UpdatePublicationForm');
  });

  it('should find the publication by id', () => {
    const performRequest = jest.fn();

    hooks.useRequestData.mockImplementationOnce(() => performRequest);

    shallow(<UpdatePublicationPage {...defaultProps} />);

    expect(hooks.useRequestData).toHaveBeenCalledWith({ wildcards: { id } });
    expect(performRequest).toHaveBeenCalled();
  });

  it('should match the snapshot', () => {
    const rendered = shallow(<UpdatePublicationPage {...defaultProps} />);

    expect(rendered).toMatchSnapshot();
  });
});
