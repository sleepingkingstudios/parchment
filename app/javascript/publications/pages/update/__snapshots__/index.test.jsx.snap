// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UpdatePublicationPage should match the snapshot 1`] = `
<Page
  breadcrumbs={<UpdatePublicationBreadcrumbs />}
  className="page-update-publication"
  layout={null}
  subtitle="5e Campaign Companion"
  title="Parchment"
>
  <h1>
    Update Publication
  </h1>
  <UpdatePublicationForm />
</Page>
`;
